org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/CompletionTests18.java

```diff
@@ -6799,4 +6799,128 @@ public void testIntersection18GH831() throws Exception {
 	assertTrue(String.format("Result doesn't contain expected method (%s)", result),
 	result.contains("get[METHOD_REF]{get(), Ljava.util.Optional<Ljava.io.Serializable;>;, ()Ljava.io.Serializable;, get, null, 60}\n"));
 }
+
+public void testGH960_onVarargArgument_expectCompletionsMatchingElementType() throws JavaModelException {
+	this.workingCopies = new ICompilationUnit[1];
+	this.workingCopies[0] = getWorkingCopy("/Completion/src/GH960.java", """
+			public class GH960 {
+				public void foo(Thread.State... states) {
```

Comment:
Is it necessary to use `Thread.State` here instead of some type defined in the test snippet?

The expected matches below are lengthy and with Java dropping backward compatibility, who knows, maybe the test will need to eventually be adjusted due to breaking changes.